apps/backend/src/users/users.service.ts

```diff
@@ -1,17 +1,96 @@
-import { Injectable } from '@nestjs/common';
+import {
+  Injectable,
+  BadRequestException,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { MongoRepository } from 'typeorm';
+import { Status } from './types';
 
 import { User } from './user.entity';
+import { UpdateUserDTO } from './users.dto';
+import { ObjectId } from 'mongodb';
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectRepository(User)
-    private usersRepository: Repository<User>,
+    private usersRepository: MongoRepository<User>,
   ) {}
 
   findAll(): Promise<User[]> {
     return this.usersRepository.find();
   }
+
+  async updateUser(
+    UpdateUserDTO: UpdateUserDTO,
```

Comment:
:pick: can we make `UpdateUserDTO` camelCase?
```suggestion
    updateUserDTO: UpdateUserDTO,
```